src/nn/graph.ts

```diff
@@ -179,23 +181,60 @@ export class MLGraph {
     for (const outputName of Object.keys(outputTensors)) {
       const tensor = outputTensors[outputName] as tf.Tensor;
       const desc = utils.createOperandDescriptorFromTensor(tensor);
-      const resource = outputs[outputName] as ArrayBufferView;
+      const resource = outputs[outputName] ;
+      utils.validateTypedArray(resource, desc.type, desc.dimensions);
+      const data = await tensor.data();
+      resource.set(data);
+      tf.dispose(tensor);
+    }
+  }
+
+  /** @internal */
+  computeSync(inputs: MLNamedArrayInputs, outputs: MLNamedArrayOutputs): void {
```

Comment:
Could you try to share code of `compute` as much as possible?

Reply:
Hi @BruceDai , any thoughts?

Reply:
Please let `compute` and `computeSync` share the code of `validateInputs`, `prepareOutputOperands` and `computeOutputTensors`. The only difference would be `await tensor.data()` and `tensor.dataSync()`.